### ISSUE TITLE
Not all subjects within a menu of the primary navigation are navigable

### ISSUE THREAD

**Issue by rautenburger** [swiss/styleguide#552]
**(Only when NVDA screen reader is enabled.)**

Not all subjects within a menu of the primary navigation is navigable using the arrow keys (nor tab).

Tested on:
https://q-src.github.io/styleguide/en/#main-navigation
(Chrome)
Tested with: NVDA screen reader

**Comment by topaxi** [comment]
I'm not sure what is going on here, and if I have correctly configured the NVDA screen reader (default settings...). Arrow keys left/right read single letters (is this normal for a screen reader?), I'm not sure which navigation points should be reachable through tabs and which should be reachable through arrow keys, maybe there are even more different ways to go through navigations which I don't know as I'm no accessibility expert (yet? :smile:). Also once I activate the screen reader, some elements magically receive new markup like `tabindex="-1"`, is this done by Firefox or the NVDA reader?

I've read up on some accessibility attributes and things and I'm not sure how to proceed.

I found someone with a similar menu structure with an implementation like this: https://stackoverflow.com/a/12279190/4737452

**Comment by rautenburger** [comment]
@almeidap 
Do you know what's up?

**Comment by bit-pro-iew-eui** [comment]
All menu points are accessible with the keyboard. It works as designed:
Tab for the main menu points
Enter for displaying the fly-out
Arrow keys for navigating inside the fly-out
Escape to close the fly-out
Closing issue.

**Comment by topaxi** [comment]
Using the NVDA screen reader, not all elements within the navigation are accessible, as @rautenburger reported. This seems like a bug?

**Comment by bit-pro-iew-eui** [comment]
The navigation menu should work with tab, instead of arrow keys.
The "X Close" should also be reachable with tab.

**Comment by gillerr** [comment]
The bootstrap-accessibility-plugin makes sure navigation sub-menu elements are accessed through arrow key down instead of tab key.

**Comment by angelogti** [comment]
For some reason the problem with NVDA remains: once a menu section is expanded, not all items are navigable with the arrows. The focus jumps from this: 

![image](https://user-images.githubusercontent.com/34123685/33757181-56db9ae2-dbf9-11e7-8abb-a1b5bd7d50e4.png)

to this, ignoring the items in between:

![image](https://user-images.githubusercontent.com/34123685/33757239-8d82527a-dbf9-11e7-9a6b-fe0c632bc03b.png)

(tested on https://admin-ch.github.io/styleguide/en/widgets.html#03-widgets-01-alerts, chrome )
